### ISSUE TITLE
preset_mode as separate entity?

### ISSUE THREAD

**Issue by martin-ef** [bosch-thermostat/home-assistant-bosch-custom-component#359]
I'm using the component on a KM200 and the connection works fine. 

There is just one thing: I would like to switch preset modes easier than to click on the three dots of the hc1 entity, which then opens a new popup window where I can see the active preset and can switch them. Since my Buderus gas heating is fairly old, I cannot use the vac modes (which is always set to auto) and the displayed temperatures. So the only thing I would really use is the preset mode. Therefore it would be nice to extract this feature into a separate entity. I tried to delete havoc-modes and min/max-temp as well as current_temperature (null) and temperature but then the entity doesn't work anymore.

Thank you very much for your help and best regards, 
Martin

**Comment by pszafer** [comment]
You can use automation with service call and call it.
You can use select entity and automation based on it which changes hvac mode.

**Comment by martin-ef** [comment]
sorry - I'm fairly new to HA and an intensive google search didn't help. 

I quite don'T understand what to do. Could you please describe in more detail?

Thank you VERY much!

**Comment by martin-ef** [comment]
Okay, I think I got it!

I created a helper (type dropdown) with three variables (home, holiday, children. 
Then I created an automation:
if => I checked whether the state of the helper changed either to home, holiday or children
then => decide between three actions, e.g. if helper condition is "home" - action is climate set preset to "Eigen1" (which is the preset mode found in climate.hc1).

hope this is correct and works.
